core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java

```diff
@@ -486,56 +488,70 @@ public void initialize(URI uri, org.apache.hadoop.conf.Configuration conf) throw
 
     Authority auth = Authority.fromString(uri.getAuthority());
     if (auth instanceof UnknownAuthority) {
-      // TODO(zac): In Alluxio 2.0 this warning will be upgraded to an exception
-      LOG.warn("Authority \"{}\" is unknown. The client will not be configured with this"
-          + " authority. The authority connection details will be loaded from your client"
-          + " configuration.",
-          auth);
-      mAlluxioHeader = getScheme() + ":///";
+      throw new IOException(String.format("Authority \"%s\" is unknown. The client will not be "
+              + "configured with this authority. The authority connection details will be loaded "
+              + "from the client configuration.", auth.toString()));
     }
 
+    mAlluxioHeader = getScheme() + ":///";
     mUri = URI.create(mAlluxioHeader);
 
     Map<String, Object> uriConfProperties = getConfigurationFromUri(uri);
+    AlluxioProperties alluxioProps = ConfigurationUtils.defaults();
```

Comment:
Can we do `mergeConfiguration` here/early? Also can `mergeConfiguration` just return an `InstancedConfiguration` that everything else can use after this point?

`mergeConfiguration` can just take the URI and hadoop conf, and internally it can merge that with the defaults and return an `InstancedConfiguration`.

Reply:
I've moved the config merge earlier and cleaned up the structure of the method